doc/development/virtualization.md

```diff
@@ -134,16 +134,40 @@ bulunan `.env` dosyası içindeki değişkenkeri doldurmanız gerekmektedir.
 
 `.env` dosyası içeriği:
 ```
-RAILS_ENV=development
-NODE_ENV=development
-RAILS_MASTER_KEY=xxxxxxxx
+RAILS_ENV=beta
```

Comment:
`beta` ile `production` arasında bir fark yok, aynı konfigürasyonu kullanıyorlar. Dolayısıyla docker, vagrant vb. yerlerde uygulamayı `beta` olarak çalıştırmamalıyız, bu çok yanlış sonuçlara ve davranışlara sebebiyet verebilir.

Reply:
Mesela bu ortamda kişi gidip fotoğraf yüklediğinde gerçek minio makinasını kirletmiş olacak. Bu basit bir örnek. Her durum için böyle sorunlar oluşacaktır. Beta veya production'a neden ihtiyaç duyuluyor baştan beri anlamadım, benzer durumları Vagrant'ta da görmüştüm, o zaman da anlamamıştım.

Reply:
Mesela production (beta) ortamında envai çeşit cache mekanizması var:

```
  config.cache_classes = true
  config.eager_load = true
  config.action_controller.perform_caching = true
```

Bu imajı kullanırken karşılaşılan bir sorunun bunlardan kaynaklanmadığını nasıl garanti edebiliriz? Edemeyiz.

Reply:
> `beta` ile `production` arasında bir fark yok, aynı konfigürasyonu kullanıyorlar. Dolayısıyla docker, vagrant vb. yerlerde uygulamayı `beta` olarak çalıştırmamalıyız, bu çok yanlış sonuçlara ve davranışlara sebebiyet verebilir.

Sunucu sistemi test etmek istiyorsak en azından sistem ekibinin, beta ya da production ortamında , yerelde test edebilmesi gerekiyor.

> Mesela bu ortamda kişi gidip fotoğraf yüklediğinde gerçek minio makinasını kirletmiş olacak. Bu basit bir örnek. Her durum için böyle sorunlar oluşacaktır. Beta veya production'a neden ihtiyaç duyuluyor baştan beri anlamadım, benzer durumları Vagrant'ta da görmüştüm, o zaman da anlamamıştım.

Evet minio'nun kirlenme tehlikesi var. beta ve production aynı minio'yu kullanıyor şu an.
beta bence kirlenebilir ama production tabii ki kirlenmemeli. Buna çözüm olarak beta'da farklı minio kullanılmalı.

Geliştirici kendi geliştirme ortamında RAILS_ENV=beta derse de aynı sorun var, vagrant ya da docker'a özel bir problem değil.

Reply:
Yanlış anlamadıysam bu docker-compose'u product ekibi için oluşturmadık mı? Eğer öyle ise product ekibinin neden uygulamayı production ortamında görmesi gerekiyor onu anlamadım.

Yalnızca minio değil mesela. Örneğin kullanıcılar oluşturulduğunda hoşgeldiniz e-mail'i atan bir kod düşün:

```ruby
unless Rails.env.development? || Rails.env.test?
  send_welcome_email
```

Bu çok tipik bir koddur. Geliştirici veya product ekibi uygulamayı beta ortamında başlatırsa tüm bu kontrolleri bypass etmiş oluyor ve tehlikeli sonuçlar ortaya çıkartabilir.

Bunun yanı sıra yalnızca minio değil mesele, tüm production ortamının credential'larına ulaşılmış oluyor.